### PR TITLE
Improve type nullability for IntervalSet

### PR DESCRIPTION
**Pull request by lppedd**
`IntervalSet` seems to be a hot path, so the less nullability checks we have, and the less iterators we use, the better.

Also tested with the ANTLR 5 test suite.

### CODE REVIEW

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt [hunk 1]
```diff
@@ -143,7 +143,7 @@ public class IntervalSet : IntSet {
     }
   }
 
-  private var _intervals: MutableList<Interval> = ArrayList(8)
+  private var _intervals: MutableList<Interval> = ArrayList(16)
```
Comment (ftomassetti) [inline antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt:146]: Should we use a constant or provide an explanation for the "magic constant"?

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt [hunk 2]
```diff
@@ -194,22 +194,18 @@ public class IntervalSet : IntSet {
       field = value
     }
 
+  // This is the most used constructor.
+  // Before, the vararg one was getting used, allocating an IntArray for no reason
+  public constructor()
```
Comment (ftomassetti) [inline antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt:199]: Should be the initialization move here?
If not, the other constructor does not first initialize _intervals and then override it with the received parameter?

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt [hunk 3]
```diff
@@ -483,27 +463,27 @@ public open class IntervalSet : IntSet {
...
 
-    val iter = intervals!!.iterator()
+    val n = _intervals.size
+    var i = 0
 
-    while (iter.hasNext()) {
-      @Suppress("LocalVariableName")
-      val I = iter.next()
+    while (i < n) {
+      val I = _intervals[i++]
```
Comment (ftomassetti) [inline antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt:486]: having `i` and capital `I` is a bit confusing. I would call this one `currInterval` and `i` -> `index`

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt [hunk 4]
```diff
@@ -544,37 +524,37 @@ public open class IntervalSet : IntSet {
...
```
Comment (ftomassetti) [inline antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt:541]: Same renaming here

**Review by ftomassetti — APPROVED** [review]
Good, but I added a few comments about minor changes

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt (hunk 2 above)
Comment (lppedd) [reply antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt]: Thanks for spotting it! Solved, see the force-pushed change.

File: antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt (hunk 3 above)
Comment (lppedd) [reply antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntervalSet.kt:491]: Yeah I know, names are not the best! But I'd like to keep it as it is to match ANTLR 4.

**Comment by ftomassetti** [comment]
Thank you @lppedd !

**Pull request merged**